client/extensions/woocommerce/app/products/product-form.js

```diff
@@ -108,3 +113,24 @@ export default class ProductForm extends Component {
 		);
 	}
 }
+
+function mapApiToProps( apiClient, ownProps, state ) {
+	const { siteId, productId } = ownProps;
+	const apiProduct = apiClient.selectors.getProduct( productId, { freshness: 30 * SECOND } );
+	const edits = getProductEdits( state, productId, siteId );
+	const product = { ...apiProduct, ...edits };
+
+	// TODO: Move this over to `siteData`
+	const variations = product && getProductVariationsWithLocalEdits( state, productId );
+
+	// TODO: Move this over to `siteData`
+	const productCategories = getProductCategoriesWithLocalEdits( state );
+
+	return {
+		product,
+		variations,
+		productCategories,
+	};
+}
+
+export default withApiClient( 'woocommerce', mapApiToProps, 'siteId' )( ProductForm );
```

Comment:
The arguments here are confusing, what do you think about converting it to an object, so they can be named?

```
{
	type: 'woocommerce',
	mapApiToProps,
	cacheKey: 'siteId', // or keyProp?
}
```

(I know I tweaked the names from your function, but I think these are more clear)

Reply:
Yeah, the argument list is getting a bit long there. I'm open to some options here. The api type and the key definitely belong together, although `mapApiToProps` itself belongs to the component. I'd like to have some further discussion, for sure.

Reply:
> The api type and the key definitely belong together, although mapApiToProps itself belongs to the component

what does this mean? Are you thinking more like an (options, callback)  pattern like this? I can see that working well:

```
export default withApiClient( {
	type: 'woocommerce',
	cacheKey: 'siteId', // or keyProp?
}, mapApiToProps )( ProductForm );
```

Reply:
I think that could work. I think `keyProp` would work fine for what we want. So, this object would be the specific client identifier.